code/processes/metrics.q

```diff
@@ -60,21 +64,13 @@ subscribe:{
 init:{
   r:subscribe[];
 
-  / make sure connection to TP was successful, or else wait
-  while[notpconnected[];
-   / wait 10 seconds
-   .os.sleep[10];
-   / try again to connect to discovery
-   .servers.startup[];
-   / try again to subscribe to TP
-   r:subscribe[]];
+  // Block process until all required processes are connected
+  r:.servers.startupdependent[requiredprocs;tpconsleep;cycles;string[subscribe];`metrics]; 
 
   / check if updates have already been sent from TP, if so recover from RDB
   if[r[`icounts][`trade] > 0;
-   / make sure connection is made to RDB
-   while[not count s:.sub.getsubscriptionhandles[`rdb;();()!()];.os.sleep[10]];
    / get handle for RDB
-   h:exec first w from s;
+   h:exec first w from s:.sub.getsubscriptionhandles[`rdb;();()!()];
    .lg.o[`recovery;"recovering ",(string r[`icounts][`trade])," records from trade table on ",string first s`procname];
    / query data from before subscription from RDB
    t:h"select time,sym,size,price from trade where i<",string r[`icounts][`trade];
```

Comment:
Bit of duplication here, if you save the string a variable on line 74 then this can be shortened to:
```
...where i<",<var>;
```